pyiron/atomistics/job/sqs.py

```diff
@@ -33,30 +36,76 @@ def pymatgen_to_pyiron(structure):
+    possible while using finite periodic cells.
+
+    A pyiron wrapper for the [SQS code of Dominik Gehringer](https://git.unileoben.ac.at/p1655622/sqsgenerator).
+
+    'Structural models used in calculations of properties of substitutionally random $A_{1-x}B_x$ alloys are usually
+    constructed by randomly occupying each of the N sites of a periodic cell by A or B. We show that it is possible to
+    design ‘‘special quasirandom structures’’ (SQS’s) that mimic for small N (even N=8) the first few, physically most
+    relevant radial correlation functions of a perfectly random structure far better than the standard technique does.
+    We demonstrate the usefulness of these SQS’s by calculating optical and thermodynamic properties of a number of
+    semiconductor alloys in the local-density formalism.'
+    From the abstract of Zunger, Wei, Ferreira, and Bernard, Phys. Rev. Lett. 65 (1990) 353,
+    DOI:https://doi.org/10.1103/PhysRevLett.65.353
+
+    Input:
+        - mole_fractions (dict): Approximate chemical composition for the output structure(s), using chemical symbols
+            as the keys and floats as the values. Values should sum to 1, but within reason will be automatically
+            adjusted to accommodate the number of atoms in the provided structure (adjustments printed to standard out).
+            Vacancies can also be included using the key '0'.
+        - weights (list/numpy.ndarray): Specifies the weights of the individual shells. (Default is None, which uses the
+            inverse of the shell number for the weight, i.e.
+            [1, 0.5, 0.3333333333, 0.25, 0.2, 0.166666667, 0.1428571429].)
+        - objective (float): Specifies the value the objective functions. The program tries to reach the specified
+            objective function. (Default is 0.)
+        - iterations (int): How many iterations to make searching for the most special quasirandom structure using a
+            random shuffling procedure. (Default is 1e6)
+        - n_output_structures (int): How many different SQS structures to return (in decreasing special
+            quasirandomness). (Default is 1.)
+    """
+
     def __init__(self, project, job_name):
         super(SQSJob, self).__init__(project, job_name)
-        self.input = GenericParameters(table_name="input")
-        self.input['mole_fractions'] = dict()
-        self.input['weights'] = None
-        self.input['objective'] = 0.0 
-        self.input['iterations'] = 1e6 
-        self.input['output_structures'] = 1
+        # self.input = InputList(table_name='input')
+        self.input = InputList(table_name='custom_dict')
+        self.input.mole_fractions = dict()
+        self.input.weights = None
+        self.input.objective = 0.0
+        self.input.iterations = 1e6
+        self.input.n_output_structures = 1
         self._python_only_job = True
         self._lst_of_struct = []
-    
+        _fail_if_imports_missing()
+        self.__hdf_version__ = "0.2.0"
```

Comment:
In addition to the `__hdf_version__` we should also define a version for the SQS code wrapper https://github.com/pyiron/pyiron_base/blob/master/pyiron_base/job/generic.py#L157 and the supported versions of the sqsgenerator https://github.com/dgehringer/sqsgenerator/tags

Reply:
Shouldn't this be handled by the conda requirements?

Reply:
I get that that there is no Windows-compliant version so we're not currently including this in the conda requirements, but do we want some more general way to handle this sort of situation? Tbh, it is not clear to me from the two links you gave what exactly you are looking for in this case.

Reply:
I mean the version that is stored in the Database, currently for python only jobs this is 0.0.1 from GenericJob and for jobs with executables the executable version is stored. Still it would be nice to use the database to filer for all old jobs so users can update them, that is why I suggested setting the version rather than just the hdf_version but I agree currently it is not consistent.